### PR TITLE
Admin create destroy edit questions, exams, fix layout to bootstrap 4, refactor

### PR DESCRIPTION
**Pull request by trantuanngoc**

### CODE REVIEW

File: app/controllers/exams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @exams = Exam.paginate(page: params[:page])
+  end
+
+  def new
+    @exam = Exam.new
+    @exam.questions.build.answers.build unless @exam.questions.build.answers.build.present?
+  end
+
+  def create
+    @exam = Exam.create(exam_params)
```
Comment (hnam2812) [inline app/controllers/exams_controller.rb:17]: Use `new`, not `create`

File: app/controllers/exams_controller.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    if @exam.save
+      flash[:success] = "Created success!"
+      redirect_to :edit
+    else
+      @subjects = Subject.all.map{ |subject| [subject.name, subject.id] }
+      flash[:error] = "Created failed!"
```
Comment (hnam2812) [inline app/controllers/exams_controller.rb:23]: show exam error

File: app/models/exam.rb [hunk 3]
```diff
@@ -1,6 +1,15 @@
 class Exam < ApplicationRecord
   has_many :user_exams
   has_many :users, through: :user_exams
-  has_many :exam_questions
-  has_many :questions, through: :exam_questions
+  belongs_to :subject
+  has_many :questions, dependent: :destroy
```
Comment (hnam2812) [inline app/models/exam.rb:5]: Split `belongs_to` and `has_many` group

File: app/models/exam.rb [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
+  validates :name, presence: true
+
+  accepts_nested_attributes_for :questions, reject_if: :all_blank, allow_destroy: true
+  validate :require_one_question
+
+  private
+    def require_one_question
```
Comment (hnam2812) [inline app/models/exam.rb:12]: indent

File: app/models/question.rb [hunk 5]
```diff
@@ -1,5 +1,13 @@
...
-  has_many :answers
+  validates :content, presence: true
+  belongs_to :exam
+  has_many :answers, dependent: :destroy
+
+  accepts_nested_attributes_for :answers, reject_if: :all_blank, allow_destroy: true
+  validate :require_two_answers
+
+  private
+    def require_two_answers
```
Comment (hnam2812) [inline app/models/question.rb:10]: indent

File: app/views/exams/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+<div class="modal-body row">
+  <div class="col-md-4">
+    <%= form_for @exam do |f| %>
+      <%= render 'shared/error_messages', resource: f.object %>
+      <%= render "exams/subject_form", f: f %>
+
+      <%= f.label :Exam %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.select :status, options_for_select([["Draft", "draft"], ["Public", "public"]]), {}, class: "selectpicker mt-3 mb-3" %>
```
Comment (hnam2812) [inline app/views/exams/edit.html.erb:11]: Use enum

File: app/views/exams/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      <%= f.submit "Save", class: "btn btn-primary mt-3 align-right" %>
+  </div>
+  <div class="col-md-8">
+
```
Comment (hnam2812) [inline app/views/exams/edit.html.erb:15]: remove

File: app/views/exams/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+      <div class="links">
+        <%= link_to_add_association 'Add', f, :questions, {data: {"association-insertion-method"=>"after"}, class: "btn btn-primary mb-2" } %>
+      </div>
+      <%= f.fields_for :questions do |question| %>
+        <%= render "question_fields", f: question%>
+      <% end %>
+
```
Comment (hnam2812) [inline app/views/exams/edit.html.erb:22]: remove

File: app/views/profiles/show.html.erb [hunk 9]
```diff
@@ -1,20 +1,19 @@
...
-        </div>
+    <div class="col-sm-12">
+      <div class="col-xs-12 col-sm-4 text-left">
+        <figure>
+          <h3>Profile</h3>
+          <%= avatar_for(@user, 100, 100) %>
+        </figure>
+      </div>
+      <div class="col-xs-12 col-sm-8">
+        <p><strong><%= (@profile.last_name||"").capitalize << (@profile.first_name||"").capitalize %> </strong></p>
```
Comment (hnam2812) [inline app/views/profiles/show.html.erb:11]: Write instance_method
```
EX:
def full_name
  "#{last_name} #{first_name}"
end

File: app/models/profile.rb [hunk 10]
```diff
@@ -1,3 +1,7 @@
 class Profile < ApplicationRecord
   belongs_to :user
+
+  def full_name
+    (self.last_name || "").capitalize << " " << (self.first_name || "").capitalize
```
Comment (vuhuutuan262-zinza) [inline app/models/profile.rb:5]: Why self

**Review by vuhuutuan262-zinza — APPROVED** [review]
LGTM

**Pull request merged**